actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -636,6 +636,8 @@ def app=(app)
         def register_encoder(*args)
           RequestEncoder.register_encoder(*args)
         end
+
+        delegate :fallback_encoder, :fallback_encoder=, to: RequestEncoder
```

Comment:
I think this would be better off being set on the `ActionDispatch::IntegrationTest` class and then in `process` we do:

```ruby
request_encoder = RequestEncoder.encoder(as || self.class.encode_requests_as)
```